### DISCUSSION TITLE
How can I wait for a full selector with "contains()" to exist?

### DISCUSSION THREAD

**Discussion by MatiDiez** [cypress-io/cypress#15875] [Questions and Help] [answered]
I'm working with dynamic tables. So When I create a new entry in the table, the row takes a couple of seconds to appear. I need to verify stuff in the new row, click buttons, etc. 

Since I have many rows already on the table, I store the new entry in a variable, and then add it to my selector within contains(): 
```
cy.get('.row').contains(newRowName) 
```

All of this works fine, the problem is that, when cypress looks for the element, it will do cy.get('.row), and since there are rows, it already finds a scope to work with. Then it will wait the 4 seconds or whatever timeout I set to find the "contains" piece but it does so, within the set it already got from the get(), where this row didn't exist. 

Result: I never finds the contains piece, because it didn't make it into the initial set. 

I resolve this of course by adding wait, something like: 

```
cy.wait(3000)
cy.get('.row').contains('.row', newRowName).find('button').click()
```

But I have already an entire minute of waits hehe. And it will keep growing. So is there a wait, for the selector timeout to retry the whole statement, and not just the chained contains() until it finds the element?

**Comment by bahmutov** [comment, marked as answer]
This is a great example of https://on.cypress.io/retry-ability and the right solution is to use `cy.contains` by itself without splitting it into `cy.get + cy.contains`

```js
// add new row, which takes a few seconds
cy.contains('.row', newRowName).find('button').click()
```
